skit.Infrastructure/Integrations/Extension.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Reflection;
+using FluentValidation;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using skit.Infrastructure.Integrations.Emails.Configuration;
+using skit.Infrastructure.Integrations.Emails.Services;
+
+namespace skit.Infrastructure.Integrations;
+
+internal static class Extension
+{
+    public static IServiceCollection AddIntegrations(this IServiceCollection services, IConfiguration configuration)
+    {
+        var smtpConfig = new SmtpConfig();
+        configuration.GetSection("SMTP").Bind(smtpConfig);
+        services.AddSingleton(smtpConfig);
```

Comment:
To jest pobieranie danych z appsettingsow? I te wszyskie pola są mapowane na obiekt SmtpConfig i dostępne z tego poziomu już w całej apce?

Reply:
Tak, potem sobie tylko w konstruktorze bierzesz tą klasę

Reply:
I ta metoda bind ma za zadanie zmapowac na ten model?